Controllers/Mobile/v1/ProfileController.cs

```diff
@@ -189,13 +190,33 @@ public async Task<IActionResult> DeleteProfilePictureAsync()
             await userService.SaveChangesAsync();
             return NoContent();
         }
+
+        [HttpGet("client-activity/{clientId}")]
+        public async Task<IActionResult> GetClientActivityByIdAsync( [FromRoute] long clientId,
+            [FromQuery] int page = 0,
+            [FromQuery] int pageSize = Constants.CLIENT_ACTIVITY_DEFAULT_PAGE_SIZE)
+        {
+            if (!ModelState.IsValid)
+                return base.CustomBadRequest();
+
+            Client? StoredId =await profileService.FindClientAsync(clientId);
+            if (StoredId == null)
+                return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "Client not found."));
+            PaginatedResult<Project> paginatedProjects = await profileService.FindClientActivitiesAsync(clientId, page, pageSize);
+            if (paginatedProjects == null)
```

Comment:
projects will not be null if the client has no projects, instead you will get an empty list